### PR TITLE
SQL: Fix issue with date columns returned always in UTC

### LINKED ISSUE #40152: SQL: Columns of type DATETIME are returned always at UTC
The timezone set by the client (REST, JDBC, ODBC) is not used when selecting columns of data type DATETIME, instead `UTC` is always used. E.g.:
```
-- timezone set to 'Etc/GMT-10'
SELECT created FROM test;

        created
------------------------
2016-01-01T00:00:00.000Z
```

The same issue (UTC is always used) occurs also for all grouping queries

### PR DESCRIPTION
**Pull request by matriv**
When selecting columns of ES type `date` (SQL's DATETIME) the
`FieldHitExtractor` was not using the timezone of the client session
but always resorted to UTC. The same behaviour (UTC only) was
encountered also for grouping keys (`CompositeKeyExtractor`) and
for First/Last functions on dates (`TopHitsAggExtractor`).

Fixes: #40152

### CODE REVIEW

**Review by costin — APPROVED** [review]
LGTM.

Please raise an another issue since in the future we should improve the way we serialize/deserialize extractors. The timezone is common property yet 3 types of extractors have a reference to it and serialize/deserialize it separately.
Ideally we would find a way to save it once per cursor and reused in all these places.

**Comment by matriv** [comment]
@costin thanks for reminding: #40216

File: x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/execution/search/extractor/CompositeKeyExtractorTests.java [hunk 1]
```diff
@@ -38,35 +38,35 @@ protected CompositeKeyExtractor createTestInstance() {
     }
 
     @Override
-    protected CompositeKeyExtractor mutateInstance(CompositeKeyExtractor instance) throws IOException {
-        return new CompositeKeyExtractor(instance.key() + "mutated", instance.property(), instance.zoneId());
+    protected CompositeKeyExtractor mutateInstance(CompositeKeyExtractor instance) {
+        return new CompositeKeyExtractor(instance.key() + "mutated", instance.property(), instance.zoneId(), instance.isDateTimeBased());
```
Comment (astefan) [inline x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/execution/search/extractor/CompositeKeyExtractorTests.java]: Is this mutation mechanism accurate here? Not talking about this PR specific change, but the method in general. Shouldn't this have 4 different mutations? (the mutated key, the property, the zone id, and the freshly added boolean) Something similar to https://github.com/elastic/elasticsearch/blob/master/x-pack/plugin/sql/sql-action/src/test/java/org/elasticsearch/xpack/sql/action/SqlQueryRequestTests.java#L102.

**Review by astefan — APPROVED** [review]
LGTM. Left two comments.

**Comment by matriv** [comment]
@elasticmachine run bwc
@elasticmachine run default-distro

**Comment by matriv** [comment]
@elasticmachine run ci/bwc
@elasticmachine run ci/default-distro

**Comment by matriv** [comment]
@elasticmachine run bwc
@elasticmachine run default-distro

**Comment by matriv** [comment]
@elasticmachine run elasticsearch-ci/bwc
@elasticmachine run elasticsearch-ci/default-distro

**Comment by matriv** [comment]
@elasticmachine run elasticsearch-ci/packaging-sample

**Comment by matriv** [comment]
Backported to `7.x` with bc4c8e53c5034d34d5322ac126ce8bcfc385f3cd
to `7.0` with 25f67ab0671c422c05b95910d2920d5d56e0f8d4
to `6.7` with 6998ea7280772816b503d72ce9a5621328848c82
to `6.6` with 3a627e1ba9bec89f4759de3d8f0ffd87678360db

**Pull request merged**